stdlib/public/stubs/GlobalObjects.cpp

```diff
@@ -53,6 +53,34 @@ swift::_SwiftEmptyArrayStorage swift::_swiftEmptyArrayStorage = {
   }
 };
 
+__asm__("  .globl __swiftImmortalRefCount\n");
+
+#if __POINTER_WIDTH__ == 64
+
+  // TODO: is there a way to avoid hard coding this constant in the inline
+  //       assembly string?
+  static_assert(swift::InlineRefCountBits::immortalBits() == 0x80000004ffffffffull,
+                "immortal refcount bits changed: correct the inline asm below");
+  __asm__(".set __swiftImmortalRefCount, 0x80000004ffffffff\n");
+
+#elif __POINTER_WIDTH__ == 32
+
+  // TODO: is there a way to avoid hard coding this constant in the inline
+  //       assembly string?
+  static_assert(swift::InlineRefCountBits::immortalBits() == 0x800004fful,
+                "immortal refcount bits changed: correct the inline asm below");
+  __asm__(".set __swiftImmortalRefCount, 0x800004ff\n");
+
+#else
+  #error("unsupported pointer width")
+#endif
+
+// Static arrays can only contain trivial elements. Therefore we can reuse
+// the metadata of the empty array buffer. The important thing is that its
+// deinit is a no-op and does not actually destroy any elements.
+__asm__("  .globl __swiftStaticArrayMetadata\n");
+__asm__(".set __swiftStaticArrayMetadata, _$ss19__EmptyArrayStorageCN\n");
```

Comment:
I'm thinking this should become the real metadata symbol, and `_$ss19__EmptyArrayStorageCN` should be an alias to it. If we're expanding its use cases, then `EmptyArrayStorage` is no longer an accurate name.

Reply:
We can do this later. It would required renaming the `EmptyArrayStorage` class. I don't want to introduce too much churn here.